src/transformers/integrations/integration_utils.py

```diff
@@ -785,20 +841,35 @@ def on_train_end(self, args, state, control, model=None, tokenizer=None, **kwarg
                     else {
                         f"eval/{args.metric_for_best_model}": state.best_metric,
                         "train/total_floss": state.total_flos,
+                        "model/num_parameters": self._wandb.config["model/num_parameters"],
                     }
                 )
+                metadata["final_model"] = True
                 logger.info("Logging model artifacts. ...")
                 model_name = (
                     f"model-{self._wandb.run.id}"
                     if (args.run_name is None or args.run_name == args.output_dir)
                     else f"model-{self._wandb.run.name}"
                 )
+                # add the model architecture to a separate text file
```

Comment:
I am not 100% sure I understand why the architecture would change over time during training?

Reply:
> I am not 100% sure I understand why the architecture would change over time during training.

@ArthurZucker : Here, we log the model architecture to a text file in the model artifact. Some of our users have requested this. This helps to audit and reproduce the experiment easily.